lib/imap.c

```diff
@@ -627,7 +627,13 @@ static CURLcode imap_perform_list(struct connectdata *conn)
     free(mailbox);
   }
 
-  if(!result)
+  const char *p = imap->custom_params;
+  while(p && *p == ' ')
+    p++;
+  if(!result && imap->custom && (strcasecompare(imap->custom, "FETCH") ||
+    (strcasecompare(imap->custom, "UID") && p && strstr(p, "FETCH") == p)))
+    state(conn, IMAP_FETCH);/* [UID] FETCH is the only cmd needs deal payload */
+  else if(!result)
```

Comment:
This feels like a pretty drastic change in behavior. Is this really necessary when you're changing the URL handing to do "right" ? Why does `FETCH` need this special case handling of `CUSTOMREQUEST` and not lots of the other imap commands?

Should this then also get explained in the documentation?

Reply:
Yes, it IS a big change because my poor codes. Sorry for my no deep into imap.c.

If enhanced, we can reduce the CUSTOMREQUEST SEARCH(for uid)-NULL-URL steps to SEARCH-FETCH.

By RFC 3501, in selected state client may customized 8 requests: CHECK, CLOSE, COPY, EXPUNGE, SEARCH, FETCH, STORE, UID. The first three have no specific response.  The others can all get whole response except FETCH/UID FETCH. 

The documentation said "When you change the request method...it will only change the actual string sent in the request." I understand URL is sending command, so does this, except we customized a non-command string.

Reply:
I don't like that amount of magic checks of the custom string. I think that goes against the idea of the string. It is not meant to be parsed and change what libcurl does.